python_modules/dagster/dagster/_grpc/types.py

```diff
@@ -405,18 +405,23 @@ class PartitionArgs(
         "_PartitionArgs",
         [
             ("repository_origin", RemoteRepositoryOrigin),
+            ("job_name", str),
+            # this is here for backcompat. it's expected to always be f"{job_name}_partition_set"
             ("partition_set_name", str),
             ("partition_name", str),
             ("instance_ref", Optional[InstanceRef]),
+            ("selected_asset_keys", Optional[AbstractSet[AssetKey]]),
         ],
     )
 ):
     def __new__(
         cls,
         repository_origin: RemoteRepositoryOrigin,
+        job_name: str,
```

Comment:
This will break when users upgrade their dagit/daemon versions without also upgrading their user code - we should make sure that we are prepared for user reports of that and maybe ensure that our docs are clear about exactly what our back-compat guarantees are in OSS - in the past we have been kind of wishy washy about it and I think we even have tests in our backcompat suite that exercise new versions of OSS with old versions of dagit

Maybe it's something we could detect in dagit and warn about so that the resulting error is less cryptic? Code servers tell us what dagster versions they are using

Reply:
> This will break when users upgrade their dagit/daemon versions without also upgrading their user code

Do you mean it will break when users upgrade their user code without upgrading their dagit/daemon versions?

Reply:
Yes sorry, I had that reversed

Reply:
> maybe ensure that our docs are clear about exactly what our back-compat guarantees are in OSS

Here's a PR that adds a section on this: https://github.com/dagster-io/dagster/pull/23734.

> Maybe it's something we could detect in dagit and warn about so that the resulting error is less cryptic?

Thoughts on where we would add this? The gRPC call that (re)loads the definitions? Do you think we should add this before merging this PR?

Reply:
I don’t think it needs to block the PR - if there’s a known exception it will emit we could catch it and add “this error may be due to version skew” in the error message.